core/server/data/migrations/versions/5.0/2022-04-08-11-46-update-mrr-for-cancelled-events.js

```diff
@@ -0,0 +1,56 @@
+const logging = require('@tryghost/logging');
+
+const {createTransactionalMigration} = require('../../utils');
+
+module.exports = createTransactionalMigration(
+    async function up(knex) {
+        const cancelledSubscriptions = await knex
+            .select(
+                'members_stripe_customers_subscriptions.id',
+                'members_stripe_customers_subscriptions.plan_currency',
+                'members_stripe_customers_subscriptions.plan_interval',
+                'members_stripe_customers_subscriptions.plan_amount'
+            )
+            .from('members_stripe_customers_subscriptions')
+            .join('members_stripe_customers', 'members_stripe_customers.customer_id', '=', 'members_stripe_customers_subscriptions.customer_id')
+            .join('members', 'members_stripe_customers.member_id', '=', 'members.id')
+            .where('members_stripe_customers_subscriptions.cancel_at_period_end', true)
+            .whereNot('members_stripe_customers_subscriptions.status', 'canceled');
+
+        if (cancelledSubscriptions.length === 0) {
+            logging.info('No cancelled subscriptions - skipping migration');
+            return;
+        }
+
+        logging.info(`Found ${cancelledSubscriptions.length} subscriptions to update events for`);
+        for (const subscription of cancelledSubscriptions) {
+            let mrrDelta;
+            if (subscription.plan_interval === 'year') {
+                mrrDelta = -1 * Math.floor(subscription.plan_amount / 12);
+            }
+            if (subscription.plan_interval === 'month') {
+                mrrDelta = -1 * subscription.plan_amount;
+            }
+            if (subscription.plan_interval === 'week') {
+                mrrDelta = -1 * subscription.plan_amount * 4;
```

Comment:
any reason to go with `* 4` rather than `* 52 / 12`?

Reply:
This is how we calculate the MRR for new events, so we're not deviating from the current practice https://github.com/TryGhost/Members/blob/main/packages/members-api/lib/repositories/member.js#L1186-L1192